storm-core/src/jvm/org/apache/storm/daemon/supervisor/ReadClusterState.java

```diff
@@ -0,0 +1,318 @@
+                supervisor.getCurrAssignment(), supervisor.getSharedContext());
+    }
+    
+    public ReadClusterState(Map<String, Object> superConf, IStormClusterState stormClusterState,
+            EventManager syncSupEventManager, String assignmentId, ISupervisor iSuper,
+            ILocalizer localizer, String host, LocalState localState,
+            IStormClusterState clusterState, AtomicReference<Map<Long, LocalAssignment>> cachedAssignments,
+            IContext sharedContext) throws Exception{
+        this.superConf = superConf;
+        this.stormClusterState = stormClusterState;
+        this.syncSupEventManager = syncSupEventManager;
+        this.assignmentVersions = new AtomicReference<Map<String, VersionedData<Assignment>>>(new HashMap<String, VersionedData<Assignment>>());
+        this.assignmentId = assignmentId;
+        this.iSuper = iSuper;
+        this.localizer = localizer;
+        this.host = host;
+        this.localState = localState;
+        this.clusterState = clusterState;
+        this.cachedAssignments = cachedAssignments;
+        
+        this.launcher = ContainerLauncher.make(superConf, assignmentId, sharedContext);
+        
+        @SuppressWarnings("unchecked")
+        List<Number> ports = (List<Number>)superConf.get(Config.SUPERVISOR_SLOTS_PORTS);
+        for (Number port: ports) {
+            slots.put(port.intValue(), mkSlot(port.intValue()));
+        }
+    }
+
+    private Slot mkSlot(int port) throws Exception {
+        Slot slot = new Slot(localizer, superConf, launcher, host, port,
+                localState, clusterState, iSuper, cachedAssignments);
+        slot.start();
+        return slot;
+    }
+    
+    @Override
+    public synchronized void run() {
+        try {
+            Runnable syncCallback = new EventManagerPushCallback(this, syncSupEventManager);
+            List<String> stormIds = stormClusterState.assignments(syncCallback);
+            Map<String, VersionedData<Assignment>> assignmentsSnapshot =
+                    getAssignmentsSnapshot(stormClusterState, stormIds, assignmentVersions.get(), syncCallback);
+            
+            Map<Integer, LocalAssignment> allAssignments =
+                    readAssignments(assignmentsSnapshot, assignmentId, readRetry);
+            if (allAssignments == null) {
+                //Something odd happened try again later
+                return;
+            }
+            Map<String, List<ProfileRequest>> topoIdToProfilerActions = getProfileActions(stormClusterState, stormIds);
+            
+            HashSet<Integer> assignedPorts = new HashSet<>();
+            LOG.debug("Synchronizing supervisor");
+            LOG.debug("All assignment: {}", allAssignments);
+            LOG.debug("Topology Ids -> Profiler Actions {}", topoIdToProfilerActions);
+            for (Integer port: allAssignments.keySet()) {
+                if (iSuper.confirmAssigned(port)) {
```

Comment:
minor: We may want to use `this.` as we do above for private vars just to be consistent in the file. Not a big deal though.